Add [s02b] Likelihood ratios and linear regression

TK5-Tim opened the pull request:
Hello everyone, 

I did the scribing for the notebook s02b. Which turned out to be the following topics: 
In the likelihood-test chapter: 
* ROC curves 
* MAP and ML estimation 
In the linear-methods chapter: 
* towards supervised learning
* sample vs population
* intro to linear regression 
* optimal weights to linear regression with least-squares 
* linear regression in Higher dimensions 

I did Create a new file for linear-methods and put everything around linear regression into the linear-methods chapter. 

Sometimes I struggled with providing additional information to the lecture notebooks or the resources, I hope that is ok but I appreciate any feedback for improvements and topics I should add. 

I put in some comments at the points I really struggled and will reiterate them here: 
- [x] In the part about ROC curves I heavily relied on PPA Chapter 2 because it was easy to understand. But please let me know if it is too similar formulated. (likelihood-tests.md line 29)
- [x] Is there any better way to highlight the lemma? I think it is fine the way I have it know but if there is a more elegant way to approach it that could help (likelihood-tests.md line 47)
- [x] There is code of the Snail example in the likelihood-tests notebook for an interactive plot that does not really work in the jupyter-book. Do you have an idea what I could put here instead or is it okay if I leave it out? (likelihood-tests.md line 66)
- [x] I was thinking of adding a section for AUC but it was not really covered in the lexture so I decided against it. (likelihood-tests.md line 67)
- [x] I think the idea of the interactive plot was different in class than how I used it. I think my way is still alright to visualize but let me know if you would prefer it in a different way. (linear-methods.md line 56)
- [x] I really struggled to put something extra meaningful into the part on deriving the optimal weights. Any suggestions what to add? I tried to work in the Recap part of 3a because with the help of that things actually made more sense.  (linear-methods.md line 113) 

I am obviously also happy for feedback on any other parts or changes that have to be made. 

The references I used besides the lecture notebooks were mainly these two lecture materials:

* [PPA Chapter 2](https://mlstory.org/)
* [Machine Learning: A Probabilistic Perspective](http://noiselab.ucsd.edu/ECE228/Murphy_Machine_Learning.pdf) 

If I should add them as references somewhere, then I can also do that but I did not see that in any of the other chapters available, so I left it out. 

Thanks, 
Tim

TK5-Tim commented:
Hey @vinith2 
I'm not able to add a reviewer myself, that's why I am notifying you about my PR. 
From personal experience I know that GitHub sometimes does not send out notifications otherwise, I don't want it to look like I am impatient.
Thanks, 
Tim

vinith2 commented on linear-methods.md:
```diff
@@ -13,6 +13,345 @@ kernelspec:
+$$
+
+Now we set the partial derivatives to zero and solve for $w_0$ and $w_1$:
+$\newcommand{\emp}{\hat{\mathbb{E}}}$
```
You are using this symbol for sample mean, I think. Can you define it as well?

vinith2 reviewed (commented):
Hey, the text looks good. Maybe for now combine the two texts. So it is easy for us to arrange it later.

TK5-Tim commented:
Hey @vinith2 
Thanks for the feedback! 

I added the snail example but unfortunately without any interactiveness in the plot. 
The slider works fine but unfortunately changing the slider does not change anything on the visualisation. probably because there is no active kernel (I assume) behind it after it is built once (like described in the blue box in the ipywidgets section of the page you linked). I still think I found a decent way to show the ROC curve of the Snail Example.

For the same reason the other plots can't be made interactive. 

I added a definition for the sample mean hopefully makes sense.

Thanks, 
Tim

vinith2 commented:
Thanks for checking it out, I will merge your PR and add your name in the headline for now. I will change this later when everything is setup
